src/javascript/app_2/Modules/Trading/Components/Form/TimePicker/trading-time-picker.jsx

```diff
@@ -2,41 +2,73 @@ import PropTypes   from 'prop-types';
 import React       from 'react';
 import { connect } from 'Stores/connect';
 import {
-    isTimeValid,
-    minDate,
+    // isTimeValid,
+    // minDate,
     setTime,
     toMoment }     from 'Utils/Date';
 import TimePicker  from 'App/Components/Form/TimePicker';
+import {
+    getSelectedTime,
+    getBoundaries } from 'Stores/Modules/Trading/Helpers/end-time';
 
 const TradingTimePicker = ({
     expiry_date,
     expiry_time,
     market_close_times,
+    market_open_times,
     onChange,
     server_time,
-    start_date,
-    start_time,
+    // start_date,
+    // start_time,
 }) => {
-    const moment_expiry                   = toMoment(expiry_date || server_time);
-    const moment_contract_start_date_time =
-        setTime(toMoment(start_date || server_time), (isTimeValid(start_time) ? start_time : server_time.format('HH:mm')));
-    const expiry_date_time                = setTime(moment_expiry.clone(), moment_contract_start_date_time.clone().add(5, 'minute').format('HH:mm'));
-    const expiry_date_market_close        = setTime(expiry_date_time.clone(), market_close_times.slice(-1)[0]);
-    const is_expired_next_day             = expiry_date_time.diff(moment_contract_start_date_time, 'day') === 1;
-    const min_date_expiry                 = moment_contract_start_date_time.clone().startOf('day');
-    const expiry_time_sessions            = [{
-        open : is_expired_next_day ? expiry_date_time.clone().startOf('day') : expiry_date_time.clone(),
-        close: is_expired_next_day ? minDate(expiry_date_time.clone().subtract(10, 'minute'), expiry_date_market_close) : expiry_date_market_close.clone(),
-    }];
+    /*
+    const contract_start = setTime(
+        toMoment(start_date || server_dt.clone()),
+        isTimeValid(start_time)
+            ? start_time
+            : server_dt.clone().format('HH:mm'));
+
+    const contract_expiry = setTime(
+        expiry_dt.clone(),
+        contract_start.clone().add(5, 'minute').format('HH:mm'));
 
+    const market_next_day_close = setTime(
+        expiry_dt.clone(),
+        market_close_times.slice(-1)[0]);
+    
+    // Boundaries for TimePicker.
+    const is_expired_next_day = contract_expiry.diff(contract_start, 'day') === 1;
+    const expiry_time_start = is_expired_next_day
+        ? contract_expiry.clone().startOf('day')
+        : contract_expiry.clone();
+    const expiry_time_end   = is_expired_next_day
+        ? minDate(
+            contract_expiry.clone().subtract(10, 'minute'),
+            market_next_day_close)
+        : market_next_day_close.clone();
+    */
```

Comment:
Why is this commented out?

Reply:
Mainly because we are not using `start_time` right now, and I want to implement it differently because I think it's quite confusing. I feel like it is trying to do 2 things at once, so rather than that, I think we should separate them, this is where the helper files come in. If we support `start_time` again, we should add the required logic in a helper file then do a check to see which modes is used and use the functions respectively, and if there is any more modes to be added, we could easily add more.

Reply:
IMO rewriting the function, if it doesn't have any part in resolving the advanced duration bug, shouldn't be done in the same PR. if it is related to fixing the bug, then it should be rewriting and removing the old code, not keeping the old code as comments.

Reply:
I am not sure if it qualifies as related to fixing the bug or not, however I rewrote it to fix the bug because the old code is too confusing and complicated (everytime I fix one bug, another bug appears). Keeping the old code is just as a reference because Morteza said we should keep the code for when we need to re-enable start time again. I think once we reimplement start time, we can remove the comments.

Reply:
I think the cleaner way would be to save these changes in either a new PR,  or in the git history and If we reimplement start time in the future -  add it back.